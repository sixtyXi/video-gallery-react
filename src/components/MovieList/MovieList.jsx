import React from 'react';
import PropTypes from 'prop-types';

import styles from './MovieList.scss';
import { MovieThumb } from '../MovieThumb/MovieThumb';

export const MovieList = ({ movies }) => {
  return movies.length ? (
    <div className={styles.listWrapper}>
      {movies.map(movie => (
        <MovieThumb key={movie.id} movie={movie} />
      ))}
    </div>
  ) : (
    <p>No results found</p>
  );
};

MovieList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object)
};

MovieList.defaultProps = {
  movies: []
};
