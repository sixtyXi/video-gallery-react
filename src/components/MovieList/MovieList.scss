.listWrapper {
  display: flex;
  flex-wrap: wrap;
  align-items: flex-start;

  article {
    margin: 20px;
  }
}

.emptyResult {
  margin: auto;
  font-size: 28px;
}
  