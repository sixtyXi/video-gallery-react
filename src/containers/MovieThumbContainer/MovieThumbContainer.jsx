import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { MovieThumb } from '../../components/MovieThumb/MovieThumb';
import { fetchMovie } from '../../actions/actions';

export const MovieThumbContainer = props => {
  const { getMovie, movie } = props;

  return <MovieThumb handleClick={() => getMovie(movie.id)} movie={movie} />;
};

MovieThumbContainer.propTypes = {
  getMovie: PropTypes.func.isRequired,
  movie: PropTypes.shape({
    id: PropTypes.number
  }).isRequired
};

const mapDispatchToProps = dispatch => ({ getMovie: id => dispatch(fetchMovie(id)) });

export default connect(
  null,
  mapDispatchToProps
)(MovieThumbContainer);
