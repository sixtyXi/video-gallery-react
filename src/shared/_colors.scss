$accent: #fd4d61;
$secondary: #414141;
