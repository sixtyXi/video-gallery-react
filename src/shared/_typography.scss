$text-light: white;
$size-m: 14px;
$size-s: 12px;
$size-xs: 10px;
$size-l: 16px;
