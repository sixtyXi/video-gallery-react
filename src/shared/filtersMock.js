export const SORT_FILTERS = [
  {
    value: 'release_date',
    title: 'release date'
  },
  {
    value: 'vote_average',
    title: 'rating'
  }
];

export const SEARCH_FILTERS = [
  {
    value: 'title',
    title: 'title'
  },
  {
    value: 'genres',
    title: 'genre'
  }
];

export const DEFAULT_FILTERS = {
  searchBy: SEARCH_FILTERS[0].value,
  sortBy: SORT_FILTERS[0].value,
  search: ''
};
