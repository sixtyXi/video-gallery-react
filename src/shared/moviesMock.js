export const MOVIES = [
  {
    id: 680,
    title: 'Pulp Fiction',
    tagline: "Just because you are a character doesn't mean you have character.",
    vote_average: 8.3,
    vote_count: 10910,
    release_date: '1994-09-10',
    poster_path: 'https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg',
    overview:
      "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
    budget: 8000000,
    revenue: 213928762,
    genres: ['Thriller', 'Crime'],
    runtime: 154
  },
  {
    id: 2,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 3,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 4,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 5,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 6,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 7,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  }
];
