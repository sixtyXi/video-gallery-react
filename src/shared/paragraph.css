.paragraph {
  height: 50px;
  display: inline-block;
  padding: 10px;
  line-height: 30px;
}
