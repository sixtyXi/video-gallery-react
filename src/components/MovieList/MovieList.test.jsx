import React from 'react';
import { shallow } from 'enzyme';

import { MovieList } from './MovieList';
import MovieThumbContainer from '../../containers/MovieThumbContainer/MovieThumbContainer';
import { MOVIES } from '../../shared/moviesMock';

describe('MovieList', () => {
  const movies = MOVIES.slice(0, 2);

  test('renders *No results found*', () => {
    const wrapper = shallow(<MovieList />);

    expect(wrapper).toMatchSnapshot();
  });

  test('renders correctly', () => {
    const wrapper = shallow(<MovieList movies={movies} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('renders two components *MovieThumbContainer*', () => {
    const wrapper = shallow(<MovieList movies={movies} />);

    expect(wrapper.find(MovieThumbContainer).length).toBe(2);
  });
});
