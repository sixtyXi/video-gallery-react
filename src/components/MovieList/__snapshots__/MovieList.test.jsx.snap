// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders *No results found* 1`] = `
<p>
  No results found
</p>
`;

exports[`MovieList renders correctly 1`] = `
<div
  className="listWrapper"
>
  <MovieThumb
    key="1"
    movie={
      Object {
        "genres": Array [
          "comedies",
        ],
        "id": 1,
        "overview": "Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.",
        "poster_path": "https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg",
        "release_date": "1995",
        "runtime": 97,
        "tagline": "Oscar-winning Movies",
        "title": "Four rooms",
        "vote_average": 3,
      }
    }
  />
  <MovieThumb
    key="2"
    movie={
      Object {
        "genres": Array [
          "action",
          "adventure",
        ],
        "id": 2,
        "overview": "From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.",
        "poster_path": "https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg",
        "release_date": "2003",
        "runtime": 150,
        "tagline": "Oscar-winning Movies",
        "title": "Kill Bill: Vol. 1",
        "vote_average": 4.5,
      }
    }
  />
</div>
`;
